### ISSUE TITLE
use_self does not diagnose tuple structs

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang/rust-clippy#3498]
[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=fdfb70343ce5f8be63662ad6467aee62):

```rust
pub struct A(i32);

impl A {
    pub fn a() -> Self {
        A(0)  // should print warning: use Self(0)
    }
}
```

does not raise any warnings when the `use_self` lint is enabled.
